pages/_app.js

```diff
@@ -1,10 +1,78 @@
+import { useState } from "react";
 import GlobalStyle from "../styles";
 
+export const maindishes = [
+  {
+    id: "1",
+    name: "Schnitzel",
+    price: 10.5,
+  },
+  {
+    id: "2",
+    name: "Spätzle",
+    price: 9.5,
+  },
+  {
+    id: "3",
+    name: "Rindergulasch",
+    price: 18.5,
+  },
+  {
+    id: "4",
+    name: "Sauerbraten",
+    price: 14.5,
+  },
+];
+
+export const drinks = [
+  {
+    id: "5",
+    name: "Apfelschorle",
+    price: 2.5,
+  },
+  {
+    id: "6",
+    name: "Johannisbeersaft",
+    price: 2.5,
+  },
+  {
+    id: "7",
+    name: "Zitroneneistee",
+    price: 3.5,
+  },
+  {
+    id: "8",
+    name: "Pfirsicheistee",
+    price: 3.5,
+  },
+];
+
 export default function App({ Component, pageProps }) {
+  const [items, setItems] = useState([]);
+  function handleAddItem(newItem) {
+    const alreadyInList = items.find((item) => item.id === newItem.id);
+    if (alreadyInList) {
+      console.log("is in list");
+      setItems(
+        items.map((item) => {
+          if (item.id !== newItem.id) {
+            return item;
+          }
+          return {
+            ...item,
+            amount: newItem.amount,
+          };
+        })
+      );
+      return;
+    }
+    console.log("not in list");
+    setItems([...items, newItem]);
+  }
```

Comment:
Now that we are passing the new amount and the id to the handler, we have to rewrite it a little bit:

```js
function handleAmountChange(id, amount) {
  const alreadyInList = items.find((item) => item.id === id);
  if (alreadyInList) {
    setItems(
      items.map((item) =>
        item.id !== id
          ? item
          : {
              ...item,
              amount: Math.max(amount, 0),
            }
      )
    );
    return;
  }
  const dish = maindishes.find((item) => item.id === id);
  setItems([...items, { ...dish, amount: Math.max(amount, 0) }]);
}
```

Observations:

- `Math.max(amount, 0)` prevents negative quantities;
- if the id is not in the cart yet, we get it from the `maindishes`;
- for the future, to support drinks as well, you may want to pass a third parameter to the handler (`itemType` or so), that you use to decide from which list of items to pick the new entry from.

Reply:
> * for the future, to support drinks as well, you may want to pass a third parameter to the handler (`itemType` or so), that you use to decide from which list of items to pick the new entry from.

yes this is a really good point. nice observation!!!


```js

function handleAmountChange(id, amount, itemType) {
  const alreadyInList = items.find((item) => item.id === id);
  if (alreadyInList) {
    setItems(
      items.map((item) =>
        item.id !== id
          ? item
          : {
              ...item,
              amount: Math.max(amount, 0),
            }
      )
    );
    return;
  }
  const dish = maindishes.find((item) => item.id === id);
  setItems([...items, { ...dish, amount: Math.max(amount, 0) }]);
}

```

so I want style the order list different, the counting could be the same. just want to have it easier to add the items clear inside the list and know what is what

```js

import ShoppingCart from "/components/shopping-cart";
import Drinks from "/components/order-menu/drink-menu";
import Maindishes from "/components/order-menu/main-menu";

export default function OrderList({ handleAmountChange, items }) {
  return (
    <>
      <h1>Order Menu</h1>
      <h2>Hauptspeisen</h2>
      <Maindishes handleAmountChange={handleAmountChange} items={items} />
      <h2>Getränke</h2>
      {/* <Drinks handleAddItem={handleAddItem} items={items} /> */}
      <ShoppingCart />
    </>
  );
}

```

and do I need the items={items} here aswell?

Reply:
Also when the selected amount is 0 it stays on the order overview, how can I remove that?